From 8d01520f301150a02bff1447c5834f3842c4a39b Mon Sep 17 00:00:00 2001
From: Dinesh Karthikeyan <quic_dinek@quicinc.com>
Date: Thu, 21 Apr 2022 15:12:04 +0530
Subject: [PATCH] ath12k: Cleanup unnecessary NULL termination check

Remove the unnecessary check for proper NULL termination while using
scnprintf since scnprintf always NULL terminates the destination
string.

Signed-off-by: Dinesh Karthikeyan <quic_dinek@quicinc.com>
---
 .../wireless/ath/ath12k/debugfs_htt_stats.c   | 477 +-----------------
 1 file changed, 1 insertion(+), 476 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c b/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c
index 16617d3..584ee61 100644
--- a/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c
+++ b/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c
@@ -62,11 +62,6 @@ static inline void htt_print_stats_string_tlv(const void *tag_buf,
 
 	len += scnprintf(buf + len, buf_len - len, "data = %s\n", data);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -181,11 +176,6 @@ static inline void htt_print_tx_pdev_stats_cmn_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "tx_active_dur_us_high = %u\n",
 			 htt_stats_buf->tx_active_dur_us_high);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -205,11 +195,6 @@ htt_print_tx_pdev_stats_urrn_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->urrn_stats, "urrn_stats",
 			   num_elems, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -229,11 +214,6 @@ htt_print_tx_pdev_stats_flush_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->flush_errs, "flush_errs",
 			   num_elems, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -253,11 +233,6 @@ htt_print_tx_pdev_stats_sifs_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->sifs_status, "sifs_status",
 			   num_elems, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -277,11 +252,6 @@ htt_print_tx_pdev_stats_phy_err_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->phy_errs, "phy_errs",
 			   num_elems, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -302,11 +272,6 @@ htt_print_tx_pdev_stats_sifs_hist_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->sifs_hist_status,
 			   "sifs_hist_status", num_elems, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -337,11 +302,6 @@ htt_print_tx_pdev_stats_tx_ppdu_stats_tlv_v(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "num_data_ppdus_ax_su_txbf = %u\n",
 			 htt_stats_buf->num_data_ppdus_ax_su_txbf);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -363,10 +323,6 @@ htt_print_tx_pdev_stats_tried_mpdu_cnt_hist_tlv_v(const void *tag_buf,
 
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->tried_mpdu_cnt_hist,
 			   "tried_mpdu_cnt_hist = %s\n", num_elements, "\n\n");
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
 
 	stats_req->buf_len = len;
 }
@@ -389,11 +345,6 @@ static inline void htt_print_hw_stats_intr_misc_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "count = %u\n",
 			 htt_stats_buf->count);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -415,11 +366,6 @@ htt_print_hw_stats_wd_timeout_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "count = %u\n",
 			 htt_stats_buf->count);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -456,11 +402,6 @@ static inline void htt_print_hw_stats_pdev_errs_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "rx_timeout_reset = %u\n",
 			 htt_stats_buf->rx_timeout_reset);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -501,11 +442,6 @@ static inline void htt_print_msdu_flow_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "current_drop_th = %u\n",
 			 htt_stats_buf->current_drop_th);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -552,11 +488,6 @@ static inline void htt_print_tx_tid_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "block_module_id = %u\n",
 			 htt_stats_buf->block_module_id);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -607,11 +538,6 @@ static inline void htt_print_tx_tid_stats_v1_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "sendn_frms_allowed = %u\n",
 			 htt_stats_buf->sendn_frms_allowed);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -641,11 +567,6 @@ static inline void htt_print_rx_tid_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "rxdesc_err_decrypt = %u\n",
 			 htt_stats_buf->rxdesc_err_decrypt);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -664,11 +585,6 @@ static inline void htt_print_counter_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "count = %u\n",
 			 htt_stats_buf->count);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -710,11 +626,6 @@ static inline void htt_print_peer_stats_cmn_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "inactive_time = %u\n",
 			 htt_stats_buf->inactive_time);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -753,11 +664,6 @@ static inline void htt_print_peer_details_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "qpeer_flags = 0x%x\n",
 			 htt_stats_buf->qpeer_flags);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -810,11 +716,6 @@ static inline void htt_print_tx_peer_rate_stats_tlv(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->tx_dcm, "tx_dcm",
 			   HTT_TX_PDEV_STATS_NUM_DCM_COUNTERS, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 
 fail:
@@ -887,11 +788,6 @@ static inline void htt_print_rx_peer_rate_stats_tlv(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->rx_pream, "rx_pream",
 			   HTT_RX_PDEV_STATS_NUM_PREAMBLE_TYPES, "\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 
 fail:
@@ -919,11 +815,6 @@ htt_print_tx_hwq_mu_mimo_sch_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "mu_mimo_ppdu_posted = %u\n",
 			 htt_stats_buf->mu_mimo_ppdu_posted);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -953,11 +844,6 @@ htt_print_tx_hwq_mu_mimo_mpdu_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "mu_mimo_ampdu_underrun_usr = %u\n",
 			 htt_stats_buf->mu_mimo_ampdu_underrun_usr);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -977,11 +863,6 @@ htt_print_tx_hwq_mu_mimo_cmn_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "hwq_id = %u\n\n",
 			 u32_get_bits(mac_id_word, HTT_TX_HWQ_STATS_HWQ_ID));
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1041,11 +922,6 @@ htt_print_tx_hwq_stats_cmn_tlv(const void *tag_buf, struct debug_htt_stats_req *
 	len += scnprintf(buf + len, buf_len - len, "txq_timeout = %u\n",
 			 htt_stats_buf->txq_timeout);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1068,11 +944,6 @@ htt_print_tx_hwq_difs_latency_stats_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->difs_latency_hist,
 			   "difs_latency_hist", data_len, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1095,11 +966,6 @@ htt_print_tx_hwq_cmd_result_stats_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->cmd_result, "cmd_result",
 			   data_len, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1121,11 +987,6 @@ htt_print_tx_hwq_cmd_stall_stats_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->cmd_stall_status,
 			   "cmd_stall_status", num_elems, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1148,11 +1009,6 @@ htt_print_tx_hwq_fes_result_stats_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->fes_result, "fes_result",
 			   num_elems, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1174,10 +1030,6 @@ htt_print_tx_hwq_tried_mpdu_cnt_hist_tlv_v(const void *tag_buf,
 			 htt_stats_buf->hist_bin_size);
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->tried_mpdu_cnt_hist,
 			   "tried_mpdu_cnt_hist", num_elements, "\n\n");
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
 
 	stats_req->buf_len = len;
 }
@@ -1199,11 +1051,6 @@ htt_print_tx_hwq_txop_used_cnt_hist_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->txop_used_cnt_hist,
 			   "txop_used_cnt_hist", num_elements, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1304,11 +1151,6 @@ static inline void htt_print_tx_sounding_stats_tlv(const void *tag_buf,
 		}
 	}
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1348,11 +1190,6 @@ htt_print_tx_selfgen_cmn_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "delayed_bar_7 = %u\n",
 			 htt_stats_buf->delayed_bar_7);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1381,11 +1218,6 @@ htt_print_tx_selfgen_ac_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "ac_mu_mimo_brpoll_3 = %u\n",
 			 htt_stats_buf->ac_mu_mimo_brpoll_3);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1430,11 +1262,6 @@ htt_print_tx_selfgen_ax_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "ax_mu_rts_trigger = %u\n",
 			 htt_stats_buf->ax_mu_rts_trigger);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1463,11 +1290,6 @@ htt_print_tx_selfgen_ac_err_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "ac_mu_mimo_brp3_err = %u\n",
 			 htt_stats_buf->ac_mu_mimo_brp3_err);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1512,11 +1334,6 @@ htt_print_tx_selfgen_ax_err_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "ax_mu_rts_trigger_err = %u\n",
 			 htt_stats_buf->ax_mu_rts_trigger_err);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1560,11 +1377,6 @@ htt_print_tx_pdev_mu_mimo_sch_stats_tlv(const void *tag_buf,
 				 "ax_ofdma_sch_nusers_%u = %u\n",
 				 i, htt_stats_buf->ax_ofdma_sch_nusers[i]);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1690,11 +1502,6 @@ htt_print_tx_pdev_mu_mimo_mpdu_stats_tlv(const void *tag_buf,
 		}
 	}
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1714,11 +1521,6 @@ htt_print_sched_txq_cmd_posted_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->sched_cmd_posted,
 			   "sched_cmd_posted", num_elements, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1738,11 +1540,6 @@ htt_print_sched_txq_cmd_reaped_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->sched_cmd_reaped,
 			   "sched_cmd_reaped", num_elements, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1765,11 +1562,6 @@ htt_print_sched_txq_sched_order_su_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->sched_order_su, "sched_order_su",
 			   sched_order_su_num_entries, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1792,11 +1584,6 @@ htt_print_sched_txq_sched_ineligibility_tlv_v(const void *tag_buf,
 			   "sched_ineligibility", sched_ineligibility_num_entries,
 			   "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1860,11 +1647,6 @@ htt_print_tx_pdev_stats_sched_per_txq_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "dur_based_sendn_term = %u\n",
 			 htt_stats_buf->dur_based_sendn_term);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1883,11 +1665,6 @@ static inline void htt_print_stats_tx_sched_cmn_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "current_timestamp = %u\n",
 			 htt_stats_buf->current_timestamp);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1908,11 +1685,6 @@ htt_print_tx_tqm_gen_mpdu_stats_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->gen_mpdu_end_reason,
 			   "gen_mpdu_end_reason", num_elements, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1933,11 +1705,6 @@ htt_print_tx_tqm_list_mpdu_stats_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->list_mpdu_end_reason,
 			   "list_mpdu_end_reason", num_elems, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -1958,11 +1725,6 @@ htt_print_tx_tqm_list_mpdu_cnt_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->list_mpdu_cnt_hist,
 			   "list_mpdu_cnt_hist", num_elems, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2039,11 +1801,6 @@ htt_print_tx_tqm_pdev_stats_tlv_v(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "sched_nonudp_notify2 = %u\n",
 			 htt_stats_buf->sched_nonudp_notify2);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2074,11 +1831,6 @@ static inline void htt_print_tx_tqm_cmn_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "desc_threshold = %u\n",
 			 htt_stats_buf->desc_threshold);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2098,11 +1850,6 @@ static inline void htt_print_tx_tqm_error_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "add_msdu_failure = %u\n",
 			 htt_stats_buf->add_msdu_failure);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2143,11 +1890,6 @@ static inline void htt_print_tx_tqm_cmdq_status_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "update_msduq_cmd = %u\n",
 			 htt_stats_buf->update_msduq_cmd);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2175,11 +1917,6 @@ htt_print_tx_de_eapol_packets_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "g2_packets = %u\n",
 			 htt_stats_buf->g2_packets);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2221,11 +1958,6 @@ htt_print_tx_de_classify_failed_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "qos_to_non_qos_in_prog = %u\n",
 			 htt_stats_buf->qos_to_non_qos_in_prog);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2306,11 +2038,6 @@ htt_print_tx_de_classify_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "mec_notify = %u\n",
 			 htt_stats_buf->mec_notify);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2342,11 +2069,6 @@ htt_print_tx_de_classify_status_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "status_invalid = %u\n",
 			 htt_stats_buf->status_invalid);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2368,11 +2090,6 @@ htt_print_tx_de_enqueue_packets_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "to_tqm_bypass = %u\n",
 			 htt_stats_buf->to_tqm_bypass);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2394,11 +2111,6 @@ htt_print_tx_de_enqueue_discard_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "is_ext_msdu = %u\n",
 			 htt_stats_buf->is_ext_msdu);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2422,11 +2134,6 @@ static inline void htt_print_tx_de_compl_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "tqm_bypass_frame = %u\n",
 			 htt_stats_buf->tqm_bypass_frame);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2447,11 +2154,6 @@ htt_print_tx_de_fw2wbm_ring_full_hist_tlv(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->fw2wbm_ring_full_hist,
 			   "fw2wbm_ring_full_hist", num_elements, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2480,11 +2182,6 @@ htt_print_tx_de_cmn_stats_tlv(const void *tag_buf, struct debug_htt_stats_req *s
 	len += scnprintf(buf + len, buf_len - len, "invalid_pdev = %u\n",
 			 htt_stats_buf->invalid_pdev);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2545,11 +2242,6 @@ static inline void htt_print_ring_if_stats_tlv(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->high_wm_hit_count,
 			   "high_wm_hit_count", HTT_STATS_HIGH_WM_BINS, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2568,11 +2260,6 @@ static inline void htt_print_ring_if_cmn_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "num_records = %u\n",
 			 htt_stats_buf->num_records);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2591,11 +2278,6 @@ static inline void htt_print_sfm_client_user_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->dwords_used_by_user_n,
 			   "dwords_used_by_user_n", num_elems, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2623,11 +2305,6 @@ static inline void htt_print_sfm_client_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "num_users = %u\n",
 			 htt_stats_buf->num_users);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2652,11 +2329,6 @@ static inline void htt_print_sfm_cmn_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "num_records = %u\n",
 			 htt_stats_buf->num_records);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2707,11 +2379,6 @@ static inline void htt_print_sring_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "internal_tail_ptr = %u\n\n",
 			 u32_get_bits(tail_ptr, HTT_SRING_STATS_INTERNAL_TAIL_PTR));
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2727,11 +2394,6 @@ static inline void htt_print_sring_cmn_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "num_records = %u\n",
 			 htt_stats_buf->num_records);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -2901,11 +2563,6 @@ static inline void htt_print_tx_pdev_rate_stats_tlv(const void *tag_buf,
 			   "tx_su_punctured_mode",
 			   HTT_TX_PDEV_STATS_NUM_PUNCTURED_MODE_COUNTERS, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 fail:
 	for (j = 0; j < HTT_TX_PEER_STATS_NUM_GI_COUNTERS; j++)
@@ -3108,11 +2765,6 @@ static inline void htt_print_rx_pdev_rate_stats_tlv(const void *tag_buf,
 	}
 	len += scnprintf(buf + len, buf_len - len, "\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 
 fail:
@@ -3186,11 +2838,6 @@ static inline void htt_print_rx_soc_fw_stats_tlv(const void *tag_buf,
 			 "target_refill_ring_recycle_cnt = %u\n",
 			 htt_stats_buf->target_refill_ring_recycle_cnt);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3211,11 +2858,6 @@ htt_print_rx_soc_fw_refill_ring_empty_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->refill_ring_empty_cnt,
 			   "refill_ring_empty_cnt", num_elems, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3237,11 +2879,6 @@ htt_print_rx_soc_fw_refill_ring_num_rxdma_err_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->rxdma_err, "rxdma_err",
 			   num_elems, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3262,11 +2899,6 @@ htt_print_rx_soc_fw_refill_ring_num_reo_err_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->reo_err, "reo_err",
 			   num_elems, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3301,11 +2933,6 @@ htt_print_rx_reo_debug_stats_tlv_v(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "last_non_zeros_sample %u\n",
 			 htt_stats_buf->last_non_zeros_sample);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3326,11 +2953,6 @@ htt_print_rx_soc_fw_refill_ring_num_refill_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->refill_ring_num_refill,
 			   "refill_ring_num_refill", num_elems, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3436,11 +3058,6 @@ static inline void htt_print_rx_pdev_fw_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "rx_recovery_reset_cnt = %u\n",
 			 htt_stats_buf->rx_recovery_reset_cnt);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3459,11 +3076,6 @@ htt_print_rx_pdev_fw_ring_mpdu_err_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->fw_ring_mpdu_err,
 			   "fw_ring_mpdu_err", HTT_RX_STATS_RXDMA_MAX_ERR, "\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3483,11 +3095,6 @@ htt_print_rx_pdev_fw_mpdu_drop_tlv_v(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->fw_mpdu_drop, "fw_mpdu_drop",
 			   num_elems, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3510,11 +3117,6 @@ htt_print_rx_pdev_fw_stats_phy_err_tlv(const void *tag_buf,
 	PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->phy_err, "phy_errs",
 			   HTT_STATS_PHY_ERR_MAX, "\n\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3542,11 +3144,6 @@ htt_print_pdev_cca_stats_hist_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len,
 			 "|  tx_frame|   rx_frame|   rx_clear| my_rx_frame|        cnt| med_rx_idle| med_tx_idle_global|   cca_obss|");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3570,11 +3167,6 @@ htt_print_pdev_stats_cca_counters_tlv(const void *tag_buf,
 			 htt_stats_buf->med_tx_idle_global_usec,
 			 htt_stats_buf->cca_obss_usec);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3614,11 +3206,6 @@ static inline void htt_print_hw_stats_whal_tx_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "sch_rx_sifs_resp_trigger= %u\n",
 			 htt_stats_buf->sch_rx_sifs_resp_trigger);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3637,11 +3224,6 @@ htt_print_pdev_stats_twt_sessions_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "num_sessions = %u\n",
 			 htt_stats_buf->num_sessions);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3678,11 +3260,6 @@ htt_print_pdev_stats_twt_session_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "sp_offset_us = %u\n",
 			 htt_stats_buf->sp_offset_us);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3700,11 +3277,6 @@ htt_print_pdev_obss_pd_stats_tlv_v(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "OBSS Tx failures PPDU = %u\n",
 			 htt_stats_buf->num_obss_tx_ppdu_failure);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3740,13 +3312,7 @@ static inline void htt_print_backpressure_stats_tlv_v(const u32 *tag_buf,
 	len += scnprintf(buf + len, buf_len - len,
 			 "============================");
 
-	if (len >= buf_len) {
-		buf[buf_len - 1] = 0;
-		stats_req->buf_len = buf_len - 1;
-	} else {
-		buf[len] = 0;
-		stats_req->buf_len = len;
-	}
+	stats_req->buf_len = len;
 }
 
 static inline void
@@ -3806,11 +3372,6 @@ htt_print_tx_selfgen_be_stats_tlv(const void *tag_buf,
 	len += scnprintf(buf + len, buf_len - len, "be_mu_rts_trigger = %u\n",
 			 htt_stats_buf->be_mu_rts_trigger);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3875,11 +3436,6 @@ htt_print_tx_selfgen_be_sched_status_stats_tlv(const void *tag_buf,
 			   HTT_TX_SELFGEN_NUM_SCH_TSFLAG_ERROR_STATS, "\n\n");
 
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3904,11 +3460,6 @@ htt_print_tx_pdev_be_ul_mu_mimo_sch_stats_tlv(const void *tag_buf,
 				   htt_stats_buf->be_ul_mu_mimo_brp_sch_nusers[i]);
 	}
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 }
 
@@ -3997,10 +3548,6 @@ htt_print_be_ul_mimo_user_stats(const void *tag_buf,
 				 htt_ul_user_stats_buf->user_index,
 				 htt_ul_user_stats_buf->be_rx_ulmumimo_mpdu_fail);
 	}
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
 
 	stats_req->buf_len = len;
 }
@@ -4106,11 +3653,6 @@ htt_print_ul_mumimo_trig_be_stats(const void *tag_buf,
 	}
 	len += scnprintf(buf + len, buf_len - len, "\n");
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 fail:
 	for (i = 0; i < HTT_TX_PEER_STATS_NUM_GI_COUNTERS; i++)
@@ -4196,10 +3738,6 @@ htt_print_be_ul_ofdma_trigger_stats(const void *tag_buf,
 			   HTT_RX_UL_MAX_UPLINK_RSSI_TRACK, "\n\n");
 
 	len += scnprintf(buf + len, buf_len - len, "\n");
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
 
 	stats_req->buf_len = len;
 fail:
@@ -4225,10 +3763,6 @@ htt_print_tx_pdev_be_dl_mu_ofdma_sch_stats_tlv(const void *tag_buf,
 				 "be_mu_ofdma_sch_nusers_%u = %u\n", i,
 				 htt_stats_buf->be_mu_ofdma_sch_nusers[i]);
 	}
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
 
 	stats_req->buf_len = len;
 }
@@ -4259,10 +3793,6 @@ htt_print_tx_pdev_be_ul_mu_ofdma_sch_stats_tlv(const void *tag_buf,
 				 "be_ul_mu_ofdma_brp_sch_nusers_%u = %u\n", i,
 				 htt_stats_buf-> be_ul_mu_ofdma_brp_sch_nusers[i]);
 	}
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
 
 	stats_req->buf_len = len;
 }
@@ -4317,11 +3847,6 @@ htt_print_tx_pdev_rate_stats_be_ofdma_tlv(const void *tag_buf,
 	for (i = 0; i < HTT_TX_PEER_STATS_NUM_GI_COUNTERS; i++)
 		kfree(tx_gi[i]);
 
-	if (len >= buf_len)
-		buf[buf_len - 1] = 0;
-	else
-		buf[len] = 0;
-
 	stats_req->buf_len = len;
 fail:
 	for (i = 0; i < HTT_TX_PEER_STATS_NUM_GI_COUNTERS; i++)
-- 
2.17.1

